multinode_integration_tests/tests/verify_bill_payment_utils/utils.rs

```diff
@@ -0,0 +1,1003 @@
+            .zip(self.serving_nodes.iter())
+            .for_each(|(expected_remaining_owed_value, serving_node)| {
+                assert_balances(
+                    &serving_node.earning_wallet,
+                    blockchain_interface,
+                    0,
+                    expected_remaining_owed_value,
+                );
+            })
+    }
+
+    fn assert_serving_nodes_addressed_received_payments(
+        &self,
+        assertions_values: &AssertionsValues,
+    ) {
+        let actually_received_payments = assertions_values
+            .final_service_fee_balances_by_serving_nodes
+            .balances;
+        let consuming_node_wallet = &self.consuming_node.consuming_wallet;
+        self.serving_nodes
+            .iter()
+            .zip(actually_received_payments.into_iter())
+            .for_each(|(serving_node, received_payment)| {
+                let original_debt = serving_node.serving_node_profile.debt_specs().balance_minor;
+                let expected_final_balance = original_debt - received_payment;
+                Self::wait_for_exact_balance_in_receivables(
+                    &serving_node.receivable_dao,
+                    expected_final_balance,
+                    consuming_node_wallet,
+                )
+            })
+    }
+
+    fn wait_for_exact_balance_in_receivables(
+        node_receivable_dao: &ReceivableDaoReal,
+        expected_value: u128,
+        consuming_node_wallet: &Wallet,
+    ) {
+        test_utils::wait_for(Some(1000), Some(15000), || {
+            if let Some(status) = node_receivable_dao.account_status(&consuming_node_wallet) {
+                status.balance_wei == i128::try_from(expected_value).unwrap()
+            } else {
+                false
+            }
+        });
+    }
+}
+
+pub const ONE_ETH_IN_WEI: u128 = 10_u128.pow(18);
+
+pub struct Ports {
+    consuming_node: u16,
+    serving_nodes: [u16; 3],
+}
+
+impl Ports {
+    pub fn new(
+        consuming_node: u16,
+        serving_node_1: u16,
+        serving_node_2: u16,
+        serving_node_3: u16,
+    ) -> Self {
+        Self {
+            consuming_node,
+            serving_nodes: [serving_node_1, serving_node_2, serving_node_3],
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct NodeAttributesCommon {
+    pub prepared_node: PreparedNodeInfo,
+    pub startup_config_opt: RefCell<Option<NodeStartupConfig>>,
+}
+
+impl NodeAttributesCommon {
+    fn new(prepared_node: PreparedNodeInfo, config: NodeStartupConfig) -> Self {
+        NodeAttributesCommon {
+            prepared_node,
+            startup_config_opt: RefCell::new(Some(config)),
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct ConsumingNode {
+    pub node_profile: ConsumingNodeProfile,
+    pub common: NodeAttributesCommon,
+    pub consuming_wallet: Wallet,
+    pub payable_dao: PayableDaoReal,
+}
+
+#[derive(Debug)]
+pub struct ServingNode {
+    pub serving_node_profile: ServingNodeProfile,
```

Comment:
You'd like to make this field name symmetric to the one used in the struct `ConsumingNode`. For that, you want to eliminate the word `serving` from it's name.